module3/lessons/error_handling.md

```diff
@@ -99,284 +99,222 @@ Take a five minutes on your own to answer the following questions:
-=> 4:   require "pry"; binding.pry
-   5:   render json: SenateMemberSerializer.new(@member)
-   6: end
-
-[1] pry(#<CongressController>)> @member
-=> #<ErrorMember:0x00007fe313dd0ee8 @code=404, @error_message="No members found with the last name: Tuyr", @status="NOT FOUND">
+Time to actually create a `poros` directory and a file for `ErrorMessage`:
 ```
-
-It looks like that's exactly what I have, so now I need to create a path within my controller for this `ErrorMember` to be serialized.
-
-Lets do this:
-
-```ruby
-class CongressController < ApplicationController
-  def search
-    @member = CongressFacade.member_by_last_name(params[:search])
-    if @member.class == SenateMember
-      render json: SenateMemberSerializer.new(@member)
-    else
-      render json: ErrorMemberSerializer.new(@member)
-    end
-  end
-
-  ...
+$ mkdir app/poros
+$ touch app/poros/error_message.rb
 ```
 
-Now, I've dream driven a serializer that I can pass my `ErrorMember` object to.
-
-Lets create that serializer file named `error_member_serializer.rb` and the corresponding class:
+The `ErrorMessage` PORO should have two attributes-- a message and a status code.
 
 ```ruby
+class ErrorMessage
+  attr_reader :message, :status_code
 
-class ErrorMemberSerializer
-  def initialize(error_object)
-    @error_object = error_object
+  def initialize(message, status_code)
+    @message = message
+    @status_code = status_code
   end
 end
-
-```
-
-Notice that we are not using the fast_jsonapi for this serializer. If I search for a senator that doesn't exist with the search parameter "tu" then I'll get some JSON based on my object:
-
-```json
-{
-  "error_object": {
-    "error_message": "No members found with the last name: tu",
-    "status": "NOT FOUND",
-    "code": 404
-  }
-}
 ```
 
-Not quite up to the standards of json:api specification. Lets create a custom method in my serailizer to format this properly:
+Now we have everything we need to finish the `ErrorSerializer#serialize_json` method.
 
 ```ruby
-class ErrorMemberSerializer
+class ErrorSerializer
   def initialize(error_object)
     @error_object = error_object
   end
 
-  def serialized_json
+  def serialize_json
     {
       errors: [
         {
-          status: @error_object.status,
-          messsage: @error_object.error_message,
-          code: @error_object.code
+          status: @error_object.status_code.to_s,
+          title: @error_object.message
         }
       ]
     }
   end
 end
 ```
 
-Now, I'll add the method call to `#serialized_json` in my controller to get the result that I want:
+When you run the sad path test, it should still be passing! This is awesome, but what if we want to rescue from the `ActiveRecord::RecordNotFound` exception in other controller actions? Adding a `rescue` block to every action that uses `Book.find(params[:id])` isn't very DRY.
+
+Enter the Rails `rescue_from` syntax. Add the following line to the top of your controller:
```

Comment:
This line "Enter the Rails `rescue_from` syntax" isn't clear the first time reading; it almost sounds like I'm supposed to enter this code somewhere. Maybe instead it could read: 
```
That's where the Rails `rescue_from` syntax comes in! `rescue_from` behaves as a Rails filter that creates a rescue for each action in this controller. 
```